code/modules/projectiles/guns/energy/special.dm

```diff
@@ -108,6 +108,91 @@ var/available_staff_transforms=list("monkey","robot","slime","xeno","human","fur
+
+/obj/item/weapon/gun/energy/staff/destruction_wand/lifekiller
+	lifekiller = 1
+
+/obj/item/weapon/gun/energy/staff/destruction_wand/process()
+	..()
+	if(power_supply.charge == power_supply.maxcharge && !lifekiller && !power_notice)
+		if(istype(src.loc, /mob/living/carbon))
+			var/mob/living/carbon/C = src.loc
+			C << "<span class='notice'>[src] pulses, full of energy.</span>"
+			power_notice = 1
+	else if(power_supply.charge < power_supply.maxcharge)
+		power_notice = 0
+
+/obj/item/weapon/gun/energy/staff/destruction_wand/attack(atom/target as mob|obj|turf|area, mob/living/user as mob, def_zone)
+	if(target == user && !mouthshoot)
+		if(!(power_supply.charge == charge_cost || lifekiller))
+			user << "<span class='warning'>[src] is not ready to fire again!</span>"
+			return
+		mouthshoot = 1
+		target.visible_message("<span class='warning'>[user] turns [src] on themself, ready to invoke its power...</span>")
+		if(!do_after(user,src, 40))
+			target.visible_message("<span class='notice'>[user] decided life was worth living</span>")
+			mouthshoot = 0
+			return
+		user.visible_message("<span class = 'warning'>[user] destroys themself!</span>")
+		playsound(user, fire_sound, 50, 1)
+		user.gib()
+		power_supply.use(charge_cost)
+		mouthshoot = 0
+		return
+	else
+		src.Fire(target,user,0,0,0)
+
+/obj/item/weapon/gun/energy/staff/destruction_wand/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	if(power_supply.charge == charge_cost || lifekiller)
+		if(istype(target, /mob/living))
+			if(!lifekiller)
+				return
+			var/mob/living/L = target
+			log_attack("[user.name] ([user.ckey]) gibbed [target] [ismob(target) ? "([target:ckey])" : ""] with [src] at ([target.x],[target.y],[target.z])[struggle ? " due to being disarmed." :""]" )
+			L.gib()
+			user.visible_message("<span class='warning'>[user] destroys [target] with [src]!</span>", \
+								 "<span class='warning'>You destroy [target] with [src]!</span>")
+			playsound(user, fire_sound, 50, 1)
+			power_supply.use(charge_cost)
+		else if(istype(target, /turf))
+			if(istype(target, /turf/simulated/wall))
+				user.visible_message("<span class='warning'>[user] erases the [target.name] with [src]!</span>", \
+									 "<span class='warning'>You erase the [target.name] with [src]!</span>")
+				playsound(user, fire_sound, 50, 1)
+				power_supply.use(charge_cost)
+				if(istype(target, /turf/simulated/wall/r_wall))
+					target.ex_act(1.0)
+				else
+					var/turf/simulated/wall/W = target
+					W.dismantle_wall(1,1)
+			else if(istype(target, /turf/simulated/floor))
+				user << "<span class='notice'>[src] fizzles quietly.</span>"
+				return
+		else
+			user.visible_message("<span class='warning'>[user] erases the [target.name] with [src]!</span>", \
+								 "<span class='warning'>You erase the [target.name] with [src]!</span>")
+			playsound(user, fire_sound, 50, 1)
+			power_supply.use(charge_cost)
+			target.Destroy()

```

Comment:
See my previous comment. I haven't looked at this code in a month, at least. It's highly possible I'm mistaken.
I probably saw Destroy() somewhere in the code, since that would be how I knew the function existed, but I don't remember exactly where. You can probably just ctrl+f through the repo if you want to be certain.